apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs

```diff
@@ -989,7 +989,128 @@ public async Task CanExecuteReadPartitionedReadCommand()
+        {
+            var builder = new SpannerConnectionStringBuilder();
+            if (!string.IsNullOrWhiteSpace(spannerToClrTypeMapping))
+            {
+                builder.SpannerToClrTypeDefaultMappings = spannerToClrTypeMapping;
+            }
+            var options = SpannerConversionOptions.ForConnectionStringBuilder(builder);
+            Value protobufValue = RunReadRequest(dbType, dbValue).Values[0];
+            // This is a hack, but simpler to do what DataReader does behind the scenes.
+            var clrValue = dbType.ConvertToClrType<object>(protobufValue, options);
+            // Check that CLR type of value is as expected.
+            Assert.Equal(clrType, clrValue.GetType());
+            // Check the values.
+            if (dbType == SpannerDbType.Float64)
+            {
+                Assert.Equal(Convert.ToDouble(dbValue).ToString(), clrValue.ToString());
+            }
+        }
+
+        private Struct RunExecuteStreamingSqlWithParameter(SpannerConnectionStringBuilder builder, SpannerParameter parameter)
+        {
+            var request = new ExecuteSqlRequest();
+            Mock<SpannerClient> spannerClientMock = SpannerClientHelpers
+                .CreateMockClient(Logger.DefaultLogger);
+            spannerClientMock
+                .SetupBatchCreateSessionsAsync()
+                .SetupExecuteStreamingSql(request);
+
+            var connection = BuildSpannerConnection(spannerClientMock, builder);
+
+            var command = connection.CreateSelectCommand("SELECT * FROM FOO");
+            command.Parameters.Add(parameter);
+            using var reader = command.ExecuteReader();
+            Assert.True(reader.HasRows);
+            spannerClientMock.Verify(client => client.ExecuteStreamingSql(
+                It.IsAny<ExecuteSqlRequest>(),
+                It.IsAny<CallSettings>()), Times.Once());
+            return request.Params;
+        }
+
+        private ListValue RunReadRequest(SpannerDbType dbType, object value)
+        {
+            var request = new ReadRequest();
+            Mock<SpannerClient> spannerClientMock = SpannerClientHelpers
+                .CreateMockClient(Logger.DefaultLogger);
+            spannerClientMock
+                .SetupBatchCreateSessionsAsync()
+                .SetupStreamingRead(request);
+
+            var connection = BuildSpannerConnection(spannerClientMock);
+
+            var command = connection.CreateReadCommand(
+                 "Foo", ReadOptions.FromColumns(new List<string> { "Col1" }),
+                 KeySet.FromParameters(new SpannerParameterCollection
+                     {
+                        { "p", dbType, value } 
+                     }));
+            using var reader = command.ExecuteReader();
+            Assert.True(reader.HasRows);
+            // Handle the case of Float64 for now.
```

Comment:
I think it would be simpler to just return the request instead of doing any validation here. Let the caller do the validation.